app/components/DocumentExplorer.tsx

```diff
@@ -258,42 +292,55 @@ function DocumentExplorer({ onSubmit, onSelect, items }: Props) {
   };
 
   const handleKeyDown = (ev: React.KeyboardEvent<HTMLDivElement>) => {
-    switch (ev.key) {
-      case "ArrowDown": {
-        ev.preventDefault();
-        setActiveNode(next());
-        break;
-      }
-      case "ArrowUp": {
-        ev.preventDefault();
-        if (activeNode === 0) {
-          focusSearchInput();
-        } else {
-          setActiveNode(prev());
-        }
-        break;
-      }
-      case "ArrowLeft": {
-        if (!searchTerm && isExpanded(activeNode)) {
-          toggleCollapse(activeNode);
-        }
-        break;
-      }
-      case "ArrowRight": {
-        if (!searchTerm) {
-          toggleCollapse(activeNode);
-        }
-        break;
-      }
-      case "Enter": {
-        if (isModKey(ev)) {
-          onSubmit();
-        } else {
-          toggleSelect(activeNode);
-        }
-        break;
-      }
+    if (ev.key === "ArrowDown" || ev.key === "ArrowUp") {
+      ev.preventDefault();
     }
+
+    const throttledOnKeyDown = throttle(
```

Comment:
Had to throttle because the event fires at a rate faster than smooth scroll animation and as a result the active node quickly runs out of view after which further scrolling via keyboard stops.

Reply:
Unfortunately it feels unresponsive and laggy now, that 100ms response time makes all the difference.It would likely feel better to remove the smooth scroll and have it be instant

Reply:
Want both the smooth scroll and 100ms throttling removed?

Reply:
I think it's worth trying yes

Reply:
Disabled! Not lagging as before, better UX I'd say :wink: